src/main/java/com/kmorarash/hwJDBC/repository/UserRepositoryJdbcImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.kmorarash.hwJDBC.repository;
+
+import com.kmorarash.hwJDBC.model.User;
+
+import java.util.Optional;
+import java.sql.*;
+
+public class UserRepositoryJdbcImpl implements UserRepository {
+    private final String jdbcUrl;
+    private final String jdbcUsername;
+    private final String jdbcPassword;
+
+    public UserRepositoryJdbcImpl(String jdbcUrl, String jdbcUsername, String jdbcPassword) {
```

Comment:
You can have JdbcUtil class and fetch env variables in that class. Using such an approach you will have better encapsulation.